server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -192,6 +204,81 @@ public void clear(String tenantId) {
         }
     }
 
+    @Override
+    public Trace getTraceFragment(String tenantId, String id) {
+        Span span = getSpan(tenantId, id);
+
+        if (span == null) {
+            return null;
+        }
+
+        InteractionNode interactionNode = spanToNode(span);
+        interactionNode.setNodes(recursiveTraceFragment(tenantId, span));
+
+        Trace trace = spanToTrace(span);
+        trace.getNodes().add(interactionNode);
+
+        return trace;
+    }
+
+
+    private List<Node> recursiveTraceFragment(String tenantId, Span parent) {
+        List<Span> spanChildren = getChildren(tenantId, SpanUniqueIdGenerator.toOriginal(parent));
```

Comment:
Not necessary needing to change, but not sure whether the 'toOriginal' util function is required. The only id that is modified would be for client spans, which we wouldn't be requesting children for. Was there a specific case you found that required this?